ballerina/commons.bal

```diff
@@ -53,6 +57,38 @@ public enum WaitingPolicy {
   LOG_AND_IGNORE
 }
 
+# Represents the configuration required to connect to a database related to task coordination.
+#
+# + host - The hostname of the database server
+# + user - The username for the database connection
+# + password - The password for the database connection
+# + port - The port number of the database server
+# + database - The name of the database to connect to
+# + options - Additional options for the MySQL connection
+# + connectionPool - The connection pool configuration
+public type DatabaseConfig record {
+  string host = "localhost";
+  string? user = ();
+  string? password = ();
+  int port = 3306;
+  string? database = ();
+  mysql:Options? options = ();
```

Comment:
Let's create `Options` record here instead of getting it from the `mysql` library?